packages/core/src/external/aws/s3.ts

```diff
@@ -152,6 +153,39 @@ export class S3Utils {
     }
   }
 
+  async doesFileExist(key: string, bucket: string): Promise<boolean> {
```

Comment:
nit: should we limit these prefixes to `is` and `has`? `fileExists()` seems pretty clear.

Reply:
I was thinking about this and the similar function, it seemed they could be written as one function to check if file exists - just w/ diff parameters. Ended up writing it to validate the idea, see if you agree and it works, please:

```ts
  async fileExists(bucket: string, key: string): Promise<boolean>;
  async fileExists(bucket: string, filters: FileExistsFilter): Promise<boolean>;
  async fileExists(bucket: string, keyOrFilters: string | FileExistsFilter): Promise<boolean> {
    if (typeof keyOrFilters === "string") {
      const fileInfo = await this.getFileInfoFromS3(keyOrFilters, bucket);
      return fileInfo.exists;
    }
    return this.filesWithPathExist({
      bucket,
      ...keyOrFilters,
    });
  }

  private async filesWithPathExist({
    bucket,
    path,
    suffix,
  }: {
    bucket: string;
    path: string;
    suffix?: string | undefined;
  }): Promise<boolean> {
    const data = await executeWithRetriesS3(() =>
      this._s3Client.send(
        new ListObjectsV2Command({
          Bucket: bucket,
          Prefix: path,
        })
      )
    );
    const contents = data.Contents;
    if (!contents) return false;

    for (let i = 0; i < contents.length; i++) {
      if (!suffix || contents[i]?.Key?.includes(suffix)) {
        return true;
      }
    }
    return false;
  }
```

Reply:
Used your version of the func with some slight changes